src/QueryHandler.cc

```diff
@@ -368,13 +370,20 @@ void QueryHandler::process_query(protobufs::queryMessage& proto_query,
 
         Json::Value& tx_responses = pmgd_query.run();
 
-        if (tx_responses.size() != root.size()) { // error
-            cmd_current = "Transaction";
-            cmd_result = tx_responses;
-            cmd_result["info"] = "Failed PMGDTransaction";
+        if (!tx_responses.isArray() || tx_responses.size() != root.size()) {
+            Json::StyledWriter writer;
+            std::cerr << "PMGD Response:" << std::endl;
+            std::cerr << writer.write(tx_responses) << std::endl;
+
+            std::string tx_error_msg("Failed PMGD Transaction");
+            if (!tx_responses.isArray() && tx_responses.isMember("info")) {
+                tx_error_msg += ": " + tx_responses["info"].asString();
+            }
+
             cmd_result["status"] = RSCommand::Error;
-            Json::StyledWriter w;
-            std::cerr << w.write(tx_responses);
+            cmd_result["info"] = tx_error_msg;
+
+            cmd_current = "Transaction";
```

Comment:
Replacing the failing command name with Transaction doesnt still help find which command failed. There is a wider change for consistency in returning error but even if you want to keep this, please shift this to another branch with the \n commit.